applications/StructuralMechanicsApplication/custom_conditions/line_load_condition.cpp

```diff
@@ -0,0 +1,458 @@
+
+    // Sizing work matrices
+    Vector pressure_on_nodes = ZeroVector( number_of_nodes );
+
+    // Pressure applied to the element itself
+    double pressure_on_condition = 0.0;
+    if (TDim == 2) {
+        if( this->Has( PRESSURE ) ) {
+            pressure_on_condition += this->GetValue( PRESSURE );
+        }
+        if( this->Has( NEGATIVE_FACE_PRESSURE ) ) {
+            pressure_on_condition += this->GetValue( NEGATIVE_FACE_PRESSURE );
+        }
+        if( this->Has( POSITIVE_FACE_PRESSURE ) ) {
+            pressure_on_condition -= this->GetValue( POSITIVE_FACE_PRESSURE );
+        }
+
+        for ( IndexType i = 0; i < pressure_on_nodes.size(); i++ ) {
+            pressure_on_nodes[i] = pressure_on_condition;
+            if( r_geometry[i].SolutionStepsDataHas( NEGATIVE_FACE_PRESSURE) ) {
+                pressure_on_nodes[i] += r_geometry[i].FastGetSolutionStepValue( NEGATIVE_FACE_PRESSURE );
+            }
+            if( r_geometry[i].SolutionStepsDataHas( POSITIVE_FACE_PRESSURE) ) {
+                pressure_on_nodes[i] -= r_geometry[i].FastGetSolutionStepValue( POSITIVE_FACE_PRESSURE );
+            }
+        }
+    }
+
+    // Vector with a loading applied to the elemnt
+    array_1d<double, 3 > line_load = ZeroVector(3);
+    if( this->Has( LINE_LOAD ) ) {
+        noalias(line_load) = this->GetValue( LINE_LOAD );
+    }
+
+    // Declaring tangent and Jacobian
+    array_1d<double, 3> tangent_xi, tangent_eta;
+    Matrix J(TDim, 1);
+
+    // Iterate over the Gauss points
+    for ( IndexType point_number = 0; point_number < integration_points.size(); point_number++ ) {
+        const double det_j = r_geometry.DeterminantOfJacobian( integration_points[point_number] );
+        const double integration_weight = GetIntegrationWeight(integration_points, point_number, det_j);
+
+        // Calculating the pressure on the gauss point
+        double gauss_pressure = 0.0;
+        for ( IndexType ii = 0; ii < number_of_nodes; ii++ ) {
+            gauss_pressure += rNcontainer( point_number, ii ) * pressure_on_nodes[ii];
+        }
+
+        // Definition of the tangent
+        if ( gauss_pressure != 0.0 ) {
+            // Definition of the tangent
+            r_geometry.Jacobian(J, point_number, integration_method);
+            GetLocalAxis1(tangent_xi, J);
+        }
+
+        // Adding contributions to the LHS matrix
+        if ( CalculateStiffnessMatrixFlag ) {
+            if ( gauss_pressure != 0.0 ) {
+                CalculateAndSubKp( rLeftHandSideMatrix, tangent_xi, rDN_De[point_number], row( rNcontainer, point_number ), gauss_pressure, integration_weight );
+            }
+        }
+        // Adding contributions to the residual vector
+        if ( CalculateResidualVectorFlag ) {
+            if ( gauss_pressure != 0.0 ) {
+                array_1d<double, 3> normal;
+
+                // Getting LOCAL_AXIS_2
+                GetLocalAxis2(tangent_eta);
```

Comment:
this may vary depending on the point (well, in 2D it is always 0 0 1)

Reply:
See implementation, it is already 0.0.1

Reply:
but in 3D it changes point to point, while this one does not

Reply:
As I said, I don't know what is expected to be defined on 3D for a beam (one singular value for the DataValueContainer will not be enough)